### ISSUE TITLE
Reuse HttpClient instance

### ISSUE THREAD

**Issue by danports** [Ackara/Plaid.NET#38]
The library instantiates a new `HttpClient` instance for every API call:
https://github.com/Ackara/Plaid.NET/blob/1acbd938732f2eaafefde0e7661da33399d497b9/src/Plaid/PlaidClient.cs#L310

This runs contrary to [Microsoft's guidance](https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client#create-and-initialize-httpclient), and I can personally attest to the fact that ignoring this guidance does lead to problems in production (not with this library specifically, but with others I used heavily). At the very least, the `HttpClient` instance should be cached for the lifetime of the `PlaidClient` instance, if not made `static`.

**Comment by viceroypenguin** [comment]
@danports  Technically, this advice is only accurate for WPF or Console applications. For long-running applications, better advice from Microsoft is given [here](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests). You can see how this is implemented in my fork here:
https://github.com/viceroypenguin/Going.Plaid.NET/blob/97b9027c4acd90c38b4ffda09a5ef1777e083a70/src/Plaid/PlaidClient.cs#L291

**Comment by Ackara** [comment]
@viceroypenguin would you like access to make commits?

**Comment by viceroypenguin** [comment]
@Ackara - the main reason I forked was due to the poor release timing. commit access does not address this concern. If you would like to give me commit access and ask me to push commits from my fork into your repo, that would be fine. However, my primary focus (vis a vis Plaid) will be on my repo, with active releases under the nuget package I've created. If you wish to remain active, feel free to do so; on the other hand, if you decide you don't want to be active, feel free to send people my way.
